lib/src/view/design_main_loopout_length.dart

```diff
@@ -0,0 +1,57 @@
+import 'dart:html';
+import 'dart:math';
+
+import 'package:over_react/over_react.dart';
+import 'package:built_collection/built_collection.dart';
+import 'package:platform_detect/platform_detect.dart';
+import 'package:scadnano/src/state/group.dart';
+import 'package:scadnano/src/view/design_main_strand_loopout.dart';
+import 'package:scadnano/src/view/transform_by_helix_group.dart';
+import 'package:tuple/tuple.dart';
+
+import '../state/helix.dart';
+import 'package:scadnano/src/state/geometry.dart';
+import '../state/strand.dart';
+import '../state/domain.dart';
+import '../state/loopout.dart';
+import 'pure_component.dart';
+import '../util.dart' as util;
+
+part 'design_main_loopout_length.over_react.g.dart';
+
+UiFactory<DesignMainLoopoutProps> DesignMainLoopout = _$DesignMainLoopout;
+
+mixin DesignMainLoopoutPropsMixin on UiProps {
+  Geometry geometry;
+  Loopout loopout;
+}
+
+class DesignMainLoopoutProps = UiProps with DesignMainLoopoutPropsMixin;
+
+class DesignMainLoopoutComponent extends UiComponent2<DesignMainLoopoutProps> with PureComponent {
+  @override
+  render() {
+    List<ReactElement> loopout_length_elts = [];
+    loopout_length_elts.add(this._loopout_length(props.loopout));
```

Comment:
There's no need to pass `props.loopout` as a parameter to a method in the same class; Just access `props.loopout` from within the method `_loopout_length`.

Sometimes I'll but the code generating the component in a separate *function* outside of the class, if I want to call it from elsewhere. (e.g., the code that creates crossovers and loopouts, since they are called by both normal strands and the view of a strand that is being dragged), and in those cases you might need to pass some props as function parameters.

But if it's a method that will only be called from within this class, don't bother passing props since they can be accessed from within the method itself.